File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    int tar = 0;
    int answer = 0;
    public int solution(int[] numbers, int target) {
        tar = target;
        dfs(1,numbers[0],numbers);
        dfs(1,numbers[0]*(-1),numbers);
        return answer;
    }

    public void dfs(int depth,int sum,int [] numbers){
        if(depth==numbers.length){
            //System.out.println(sum);
            if(sum==tar)answer++;
            return;
        }
        dfs(depth+1,sum+numbers[depth],numbers);
        dfs(depth+1,sum+numbers[depth]*(-1),numbers);
    }
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int stopSize;
int tar;
int ans;
void dfs(vector<int>nums,int sum, int depth){
    //cout<<sum<<" "<<depth<<"\n";
    if(depth==stopSize){
        if(sum==tar)ans++;
        return;
    }
    dfs(nums,sum+nums[depth],depth+1);
    dfs(nums,sum-nums[depth],depth+1);
}

int solution(vector<int> numbers, int target) {
    stopSize=numbers.size();
    //cout<<stopSize<<"\n";
    tar=target;
    dfs(numbers,numbers[0],1);
    dfs(numbers,numbers[0]*(-1),1);
    return ans;
}